src/core/thread/mle.hpp

```diff
@@ -301,6 +301,9 @@ class Header
         kCommandAnnounce             = 15, ///< Announce
         kCommandDiscoveryRequest     = 16, ///< Discovery Request
         kCommandDiscoveryResponse    = 17, ///< Discovery Response
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
```

Comment:
Avoid wrapping these kinds of definitions to ensure that there are no symbol redefinitions even when `OPENTHREAD_CONFIG_ENABLE_TIME_SYNC` is 0.  I would also add to the comment that these are only used with `OPENTHREAD_CONFIG_ENABLE_TIME_SYNC`.